### PR TITLE
fix #9176: `const a = gorgeEx(...)` = `let a = gorgeEx(...)`; ditto w other procs

### LINKED ISSUE #9176: `echo gorgeEx("unexistant") != const a = gorgeEx("unexistant"); echo a`
```nim
echo gorgeEx("unexistant")
const a = gorgeEx("unexistant")
echo a
```

(output: "", exitCode: 0)
(output: "/bin/sh: unexistant: command not found", exitCode: 127)

### PR DESCRIPTION
**Pull request by timotheecour**
* fixes https://github.com/nim-lang/Nim/issues/9176

/cc @Araq @ianmcxa
~~* depends on https://github.com/nim-lang/Nim/pull/9925 (when viewing diff)~~ (that was merged)

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -24,6 +24,8 @@
 - The `not` operator is now always a unary operator, this means that code like
   ``assert not isFalse(3)`` compiles.
 
+- `getProjectPath`, `gorgeEx` now return the same at RT or CT (before, they returned
```
Comment (Araq) [inline changelog.md]: What is RT or CT? (I know...) Keep the jargon to a minimum. Not everybody is expert and many people even don't want to be experts.

**Comment by Araq** [comment]
I don't understand why this annotation changes anything.

**Comment by timotheecour** [comment]
PTAL

> I don't understand why this annotation changes anything.

well IMO we should improve the compiler so that bugs like https://github.com/nim-lang/Nim/issues/9176 can't happen (ie you'd get a CT error instead of `""`), but in the meantime, this fixes the bug for the procs I patched.

**Comment by timotheecour** [comment]
ping @Araq

**Comment by Araq** [comment]
Let's instead make the compiler produce an error.

**Pull request closed without merging**